src/java.base/share/classes/javax/crypto/EncryptedPrivateKeyInfo.java

```diff
@@ -304,15 +315,167 @@ private PKCS8EncodedKeySpec getKeySpecImpl(Key decryptKey,
+        }
+
+        AlgorithmId algId;
+        byte[] encryptedData;
+        DerOutputStream out = new DerOutputStream();
+        try {
+            cipher.init(Cipher.ENCRYPT_MODE, skey, params);
+            encryptedData = cipher.doFinal(key.getEncoded());
+            algId = new AlgorithmId(Pem.getPBEID(algorithm),
+                cipher.getParameters());
+            algId.encode(out);
+            out.putOctetString(encryptedData);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new IllegalArgumentException(e);
+        } catch (IllegalBlockSizeException | BadPaddingException |
+                 InvalidKeyException e) {
+            throw new SecurityException(e);
+        }
+        return new EncryptedPrivateKeyInfo(
+            DerValue.wrap(DerValue.tag_Sequence, out).toByteArray(),
+            encryptedData, algId, cipher.getParameters());
+    }
+
+    /**
+     * Creates and encrypts an `EncryptedPrivateKeyInfo` from a given PrivateKey
+     * and password.
+     *
+     * @implNote The encryption uses the algorithm set by `jdk.epkcs8.defaultAlgorithm`
+     * Security Property by the default provider and default the
+     * AlgorithmParameterSpec of that provider.
+     *
+     * @param key The PrivateKey object to encrypt.
+     * @param password the password used in the PBE encryption.  This array is
+     *                 cloned before being used.
+     * @return an EncryptedPrivateKeyInfo.
+     * @throws IllegalArgumentException on initialization errors based on the
+     * arguments passed to the method.
+     * @throws SecurityException on a encryption errors.
+     * @throws NullPointerException when the password is null.
+     *
+     * @since 24
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.PEM_API)
+    public static EncryptedPrivateKeyInfo encryptKey(PrivateKey key,
+        char[] password) {
+        char[] pass = password.clone();
+        if (Pem.DEFAULT_ALGO == null || Pem.DEFAULT_ALGO.length() == 0) {
+            throw new SecurityException("Security property " +
+                "\"jdk.epkcs8.defaultAlgorithm\" may not specify a " +
+                "valid algorithm.  Operation cannot be performed.");
+        }
+        return encryptKey(key, pass, Pem.DEFAULT_ALGO,
+            null, null);
+    }
+
+    /**
+     * Return a PrivateKey from the encrypted data in the object.
+     *
+     * @param password the password used in the PBE encryption.
+     * @return a PrivateKey
+     * @throws InvalidKeyException if an error occurs during parsing of the
+     * encrypted data or creation of the key object.
+     *
+     * @since 24
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.PEM_API)
+    public PrivateKey getKey(char[] password) throws InvalidKeyException {
+        return getKey(password, null);
+    }
+
+    /**
+     * Return a PrivateKey from the object's encrypted data with a KeyFactory
```

Comment:
Should be `SecretKeyFactory`. Same in `@param provider` below.

Also, "Return a key" is not clear. Existing `getKeySpec` methods use "Extract". You maybe also use "Recover".

Reply:
I think "Return" is very clear and I'm not a fan of "Extract" or "Recover".   The other methods are overly descriptive of the method's internals rather than the just stating the purpose.  Even with that, the other methods end with "returns it."